apps/studio/src/server/modules/user/user.router.ts

```diff
@@ -0,0 +1,282 @@
+              ...user,
+              email: user.email.toLowerCase(),
+              siteId,
+              trx,
+            })
+            return {
+              id: createdUser.id,
+              email: createdUser.email,
+              role: resourcePermission.role,
+            }
+          }),
+        )
+      })
+
+      // TODO: Send welcome email to users
+      // Email service is not ready yet
+
+      return createdUsers
+    }),
+
+  delete: protectedProcedure
+    .input(deleteInputSchema)
+    .output(deleteOutputSchema)
+    .mutation(async ({ ctx, input: { siteId, userId } }) => {
+      await validatePermissionsForManagingUsers({
+        siteId,
+        userId: ctx.user.id,
+        action: "delete",
+      })
+
+      if (userId === ctx.user.id) {
+        throw new TRPCError({
+          code: "FORBIDDEN",
+          message: "You cannot delete your own account",
+        })
+      }
+
+      const userToDeletePermissionsFrom = await db
+        .selectFrom("User")
+        .where("id", "=", userId)
+        .where("deletedAt", "is", null)
+        .selectAll()
+        .executeTakeFirst()
+
+      if (!userToDeletePermissionsFrom) {
+        throw new TRPCError({
+          code: "NOT_FOUND",
+          message: "User not found",
+        })
+      }
+
+      const deletedUserPermissions = await db
+        .updateTable("ResourcePermission")
+        .where("userId", "=", userId)
+        .where("siteId", "=", siteId)
+        .where("deletedAt", "is", null)
+        .set({ deletedAt: new Date() })
+        .returningAll()
+        .executeTakeFirst()
+
+      if (!deletedUserPermissions) {
+        throw new TRPCError({
+          code: "NOT_FOUND",
+          message: "User permissions not found",
+        })
+      }
+
+      return {
+        id: userToDeletePermissionsFrom.id,
+        email: userToDeletePermissionsFrom.email,
+      }
+    }),
+
+  getUser: protectedProcedure
+    .input(getUserInputSchema)
+    .output(getUserOutputSchema)
+    .query(async ({ ctx, input: { siteId, userId } }) => {
+      await validatePermissionsForManagingUsers({
+        siteId,
+        userId: ctx.user.id,
+        action: "read",
+      })
+
+      const result = await getUsersQuery({ siteId, getIsomerAdmins: false })
+        .where("ActiveUser.id", "=", userId)
+        .select((eb) => [
+          "ActiveUser.id",
+          "ActiveUser.email",
+          "ActiveUser.name",
+          "ActiveUser.lastLoginAt",
+          eb.ref("ActiveResourcePermission.role").as("role"),
+        ])
+        .executeTakeFirst()
+
+      if (!result) {
+        throw new TRPCError({
+          code: "NOT_FOUND",
+          message: "User not found",
+        })
+      }
+
+      return result
+    }),
+
+  list: protectedProcedure
+    .input(listInputSchema)
+    .output(listOutputSchema)
+    .query(
+      async ({ ctx, input: { siteId, getIsomerAdmins, offset, limit } }) => {
```

Comment:
do we have default values for the offset, limit?

Reply:
the input schema is importing from existing `offsetPaginationSchema` which uses default 10 as limit